Mathlib/NumberTheory/Zagier.lean

```diff
@@ -0,0 +1,288 @@
+/-
+Copyright (c) 2023 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.GroupTheory.PGroup
+
+/-!
+# Zagier's "one-sentence proof" of Fermat's theorem on sums of two squares
+
+"The involution on a finite set `S = {(x, y, z) : ℕ × ℕ × ℕ | x ^ 2 + 4 * y * z = p}` defined by
```

Comment:
```suggestion
"The involution on the finite set `S = {(x, y, z) : ℕ × ℕ × ℕ | x ^ 2 + 4 * y * z = p}` defined by
```
we might as well cite verbatim, see https://people.mpim-bonn.mpg.de/zagier/files/doi/10.2307/2323918/fulltext.pdf